recipes-core/varstored/varstored/0001-add-oxtdb-varstore.patch

```diff
@@ -0,0 +1,891 @@
++        free(path);
++        return string;
++    } else {
++        free(path);
++        return NULL;
++    }
++}
++
++/* The VM's uuid. Used for saving to the db. */
++char *oxtdb_arg_uuid;
++
++size_t
++oxtdb_serialize_variables(uint8_t **out, size_t *len, bool only_nv)
++{
++    return xapidb_serialize_variables(out, len, only_nv);
++}
++
++static bool
++write_all(const char *buf)
++{
++    bool check = false;
++    int var_num = 0;
++    check = oxtdb_exists("/vm/%s/varstored-efivar-blob/%d", oxtdb_arg_uuid, var_num);
++    if (check) {
++        char *temp = oxtdb_read("/vm/%s/varstored-efivar-blob/%d", oxtdb_arg_uuid, var_num);
++        if (strcmp(buf, temp) == 0) {
++            free(temp);
++            ERR("found match in write");
++            return false;
++        }
++        if (temp != NULL) {
++            free(temp);
++        }
++    }
++    oxtdb_write((char *)buf, "/vm/%s/varstored-efivar-blob/%d", oxtdb_arg_uuid, var_num);
++    return true;
++}
++
++static void
++read_all(char **buf)
++{
++    int var_num = 0;
++    bool ret = false;
++
++    ret = oxtdb_exists("/vm/%s/varstored-efivar-blob/%d", oxtdb_arg_uuid, var_num);
++    if (ret) {
++        DBG("attempting to cat %d\n", var_num);
++        *buf = oxtdb_read("/vm/%s/varstored-efivar-blob/%d", oxtdb_arg_uuid, var_num);
++    } else {
++        *buf = NULL;
++    }
++}
++
++bool
++oxtdb_set_variable(void)
++{
++    uint8_t *buf;
++    char *encoded;
++    size_t len;
++    bool ret = false;
++
++    if (!oxtdb_arg_uuid)
++        return true;
++
++    oxtdb_serialize_variables(&buf, &len, true);
++    if (len == 0)
++        return false;
++
++    if (!base64_encode(buf, len, &encoded)) {
++        free(buf);
++        return false;
++    }
++    free(buf);
++
++    // DoS protection
++    maybe_sleep();
++
++    ret = write_all(encoded);
++    free(encoded);
++
++    return ret;
++}
++
++bool
++oxtdb_parse_blob(uint8_t **buf, int len) {
++    return xapidb_parse_blob(buf, len);
++}
++
++enum backend_init_status
++oxtdb_init(void)
++{
++    char *encoded;
++    uint8_t *buf, *ptr;
++    BIO *bio, *b64;
++    int max_len, n, total = 0;
++    int ret = 0;
++    int var_num = 0;
++
++    if (!oxtdb_exists("/vm/%s/varstored-efivar-blob/%d", oxtdb_arg_uuid, var_num)) {
++       oxtdb_write("firstboot","/vm/%s/varstored", oxtdb_arg_uuid);
++    }
++
++    DBG("Made into oxtdb_init /vm/%s/varstored\n", oxtdb_arg_uuid);
++    char *check = oxtdb_read("/vm/%s/varstored", oxtdb_arg_uuid);
++
++    DBG("check %s\n", check);
++    if(strcmp(check, "true") == 0) {
++        DBG("Data in init db_read\n");
++        ret = 1;
++    } else if (strcmp(check, "firstboot") == 0){
++        INFO("Backend init firstboot\n");
++        oxtdb_write("true","/vm/%s/varstored", oxtdb_arg_uuid);
++        free(check);
++        return BACKEND_INIT_FIRSTBOOT;
++    } else {
++        free(check);
++        return BACKEND_INIT_FAILURE;
++    }
++    free(check);
++    read_all(&encoded);
```

Comment:
This seems like it could just be:
```
if (!oxtdb_exists("/vm/%s/varstored-efivar-blob/%d", oxtdb_arg_uuid, var_num)) {
    return BACKEND_INIT_FIRSTBOOT;
}

read_all(&encoded);
```
With the use of "/vm/%s/varstored" removed.  Unless I'm missing something.